apps/authentication/routes.py

```diff
@@ -241,6 +243,94 @@ def resend_code():
+                f"{confirmation_token}\n\n"
+                "If you have not recently tried to sign up into HackInSDN, you can ignore this e-mail.\n\n"
+                "--\n\n"
+                f"You're receiving this email because of your account on Dashboard HackInSDN."
+            ),
+            html=render_template('mail/confirmation_token.html', confirmation_token=confirmation_token),
+        )
+        mail.send(msg)
+
+        return redirect(url_for('authentication_blueprint.confirm_reset_password'))
+
+    else:
+        return render_template('pages/reset_password.html', form=form )
+
+@blueprint.route('/confirm-reset-password', methods=['GET', 'POST'])
+def confirm_reset_password():
+    form = ResetPasswordConfirmForm(request.form)
+
+    confirmation_token = session.get('confirmation_token')
+
+    if not confirmation_token or not session.get('user') or not session.get('datetime') or session.get('try') == None:
+        db.session.add(LoginLogging(ipaddr=get_remote_addr(), login=session.get('user'), auth_provider="local", success=False, message="Invalid session parameters"))
+        db.session.commit()
+        return redirect(url_for('authentication_blueprint.reset_password'))
+    
+   
+    if 'confirm' in request.form:
+        user = Users.query.filter_by(email=session.get('user')).first()
+        created_at = session.get('datetime')
+
+        if session.get('try') > 3:
+            db.session.add(LoginLogging(ipaddr=get_remote_addr(), login=session.get('user'), auth_provider="local", success=False, message="Too many tries"))
+            db.session.commit()
+            session.pop('confirmation_token')
+            session.pop('user')
+            session.pop('datetime')
+            session.pop('try')
+            return render_template('pages/confirm_reset_password.html', msg='Too many tries, please <a href=/reset-password>click here</a> to reset your password again', success=False, form=form)
+            
+     
+        now = utcnow()
+        if now - created_at > timedelta(minutes=5):
+            return render_template('pages/confirm_reset_password.html', msg='Token expired, please <a href=/reset-password>click here</a> to reset your password again', success=False, form=form)
+
+        if request.form['confirmation_token'] != confirmation_token:
+            db.session.add(LoginLogging(ipaddr=get_remote_addr(), login=session.get('user'), auth_provider="local", success=False, message="Invalid token"))
+            db.session.commit()
+            # Increment try
+            session['try'] += 1
+            return render_template('pages/confirm_reset_password.html', msg='Invalid token', success=False, form=form)
+
+        session.pop('confirmation_token')
+        session.pop('user')
+        session.pop('datetime')
+        session.pop('try')
+
+        user.password = hash_pass(request.form['password'])
+        db.session.add(user)
```

Comment:
No need to add here. When you are changing the SQLAlchemy already have that object under control